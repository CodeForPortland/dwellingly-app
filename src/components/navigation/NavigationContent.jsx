import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { injectIntl, intlShape } from 'react-intl';
import AdminMenu from './AdminMenu';
import StaffMenu from './StaffMenu';
import { ROLES } from '../../constants/constants';
import { getUserRoleString } from '../../utils';
// import { NAVIGATION } from '../../translations/messages';

class NavigationContent extends Component {
  render() {
    const { handleToggleMenu, intl, type, user } = this.props;
    const roleStr = getUserRoleString(user.role, ROLES);
    return (
      <nav className={`menu${type && ` menu--${type}`}`}>
        {user &&
          type === 'mobile' &&
          handleToggleMenu && (
            <button
              type="button"
              aria-label="Close Menu"
              className="menu__bg"
              onClick={handleToggleMenu}
            />
          )}
        <div className="menu__groups">
          {user &&
            roleStr === ROLES.STAFF && (
              <StaffMenu
                handleToggleMenu={handleToggleMenu}
                intl={intl}
                type={type}
                user={user}
              />
            )}
          {user &&
            roleStr === ROLES.ADMIN && (
              <AdminMenu
                handleToggleMenu={handleToggleMenu}
                intl={intl}
                type={type}
                user={user}
              />
            )}
        </div>
      </nav>
    );
  }
}

NavigationContent.propTypes = {
  intl: intlShape.isRequired,
  handleToggleMenu: PropTypes.func,
  type: PropTypes.string,
  user: PropTypes.shape({})
};

NavigationContent.defaultProps = {
  handleToggleMenu: undefined,
  type: 'mobile',
  user: undefined
};

export default injectIntl(NavigationContent);
