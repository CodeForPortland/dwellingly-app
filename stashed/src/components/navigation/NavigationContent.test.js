import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { IntlProvider } from 'react-intl';
import NavigationContent from './NavigationContent';

describe('NavigationContent Component', () => {
  it('renders button when type=mobile', () => {
    const props = {
    type: 'mobile',
    user: {},
    handleToggleMenu: jest.fn,
  }
    const NavigationContentComponent = renderer.create(
      <IntlProvider>
        <MemoryRouter>
          <NavigationContent {...props} />
        </MemoryRouter>
      </IntlProvider>
      ).toJSON();
    expect(NavigationContentComponent).toMatchSnapshot();
  });
});
