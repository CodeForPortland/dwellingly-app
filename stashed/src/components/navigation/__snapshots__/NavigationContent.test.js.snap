// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationContent Component renders button when type=mobile 1`] = `
<nav
  className="menu menu--mobile"
>
  <button
    aria-label="Close Menu"
    className="menu__bg"
    onClick={[Function]}
    type="button"
  />
  <div
    className="menu__groups"
  />
</nav>
`;
